crucible-llvm/src/Lang/Crucible/LLVM/Translation.hs

```diff
@@ -152,298 +159,36 @@ import           Lang.Crucible.Types
 
 type ModuleCFGMap arch = Map L.Symbol (C.AnyCFG (LLVM arch))
 
+-- | A @GlobalInitializerMap@ records the initialized values of globals in an @L.Module@.
+--
+-- The @Left@ constructor is used to signal errors in translation,
+-- which can happen when:
+-- * The global isn't actually a compile-time constant
+-- * The declaration is ill-typed
+-- * The global isn't linked ("extern global")
+--
+-- These failures are as granular as possible (attached to the values)
+-- so that simulation still succeeds if the module has a bad global that the
+-- verified function never touches.
+--
+-- To actually initialize globals, saw-script translates them into
+-- instances of @MemModel.LLVMVal@.
+type GlobalInitializerMap = Map L.Symbol (Either String LLVMConst)
+
 -- | The result of translating an LLVM module into Crucible CFGs.
 data ModuleTranslation arch
    = ModuleTranslation
-      { cfgMap :: ModuleCFGMap arch
-      , initMemoryCFG :: C.SomeCFG (LLVM arch) EmptyCtx UnitType
+      { cfgMap        :: ModuleCFGMap arch
       , _transContext :: LLVMContext arch
+      , globalMap     :: GlobalInitializerMap
```

Comment:
This should be renamed to `globalInitMap`, it conflicts with `Lang.Crucible.Simulator.SymGlobalState.globalMap`.